### ISSUE TITLE
ML Model Training Pipeline

### ISSUE THREAD

**Issue by lawrab** [lawrab/haar-weather#25]
## Description
Implement LightGBM/RandomForest training pipeline.

## Tasks
- [ ] Create models/ml.py
- [ ] Implement model training function
- [ ] Hyperparameter tuning (GridSearchCV or Optuna)
- [ ] Cross-validation strategy (temporal splits)
- [ ] Model serialization (joblib/pickle)
- [ ] Store model metadata in model_runs table
- [ ] Track feature importance
- [ ] Command: `haar train`

## Models to Try
- RandomForestRegressor (baseline)
- LightGBM (primary)
- Optional: XGBoost, gradient boosting

## Target Variables
- temperature_c (primary)
- precipitation_mm
- wind_speed_ms

## Definition of Done
- Can train models via CLI
- Models saved with versioning
- Hyperparameters tunable
- Training metadata stored in database

**Comment by lawrab** [comment]
## Implementation Plan

See full details: `docs/ml-pipeline-implementation.md`

### Approach: MOS-Style Statistical Post-Processing
Rather than building a pure ML weather model, we build a Model Output Statistics (MOS) correction system that:
1. Takes NWP forecast (ECMWF/GFS/ICON) as primary input
2. Adds local observations to capture current state
3. Learns systematic NWP biases for this location

### Files to Create
- `haar/features/temporal.py` - Cyclical time encoding, observation lags
- `haar/features/nwp.py` - NWP forecast feature extraction
- `haar/features/pipeline.py` - Feature orchestration
- `haar/models/ml.py` - LightGBM/RandomForest training
- `haar/models/baseline.py` - Persistence & climatology baselines
- `haar/models/evaluation.py` - Metrics & skill scores

### CLI Command
```bash
haar train run --target temperature_c --algorithm lightgbm --allow-limited-data
haar train run --target all  # trains all variables × all lead times
```

### Key Classes
- `FeaturePipeline` - Builds training datasets from observations + NWP forecasts
- `WeatherModelTrainer` - Handles training, CV, model serialization
- `TrainingResult` - Stores model, metrics, feature importance

### Features Used
1. **NWP Features**: ECMWF/GFS/ICON forecasts for target time
2. **Temporal**: hour_sin/cos, day_of_year_sin/cos, is_daytime
3. **Observation Lags**: 1h, 3h, 6h, 12h, 24h lookback
4. **Derived**: pressure_tendency, temp_change, wind u/v components

### Data Note
Currently ~6 days of data vs 90-day minimum. Adding `allow_limited_data` config flag for dev mode. For production, run `haar collect historical --days 365` to backfill ERA5 data.
